src/execution/operator/helper/physical_limit_percent.cpp

```diff
@@ -0,0 +1,127 @@
+};
+
+unique_ptr<GlobalSinkState> PhysicalLimitPercent::GetGlobalSinkState(ClientContext &context) const {
+	return make_unique<LimitPercentGlobalState>(*this);
+}
+
+SinkResultType PhysicalLimitPercent::Sink(ExecutionContext &context, GlobalSinkState &gstate, LocalSinkState &lstate,
+                                          DataChunk &input) const {
+	D_ASSERT(input.size() > 0);
+	auto &state = (LimitPercentGlobalState &)gstate;
+	auto &limit_percent = state.limit_percent;
+	auto &offset = state.offset;
+
+	// get the next chunk from the child
+	if (!state.is_limit_percent_delimited) {
+		Value val = PhysicalLimit::GetDelimiter(input, limit_expression.get());
+		if (!val.is_null) {
+			limit_percent = val.GetValue<double>();
+		}
+		if (limit_percent < 0.0) {
+			throw BinderException("Percentage value(%f) can't be negative", limit_percent);
+		}
+		state.is_limit_percent_delimited = true;
+	}
+	if (!state.is_offset_delimited) {
+		Value val = PhysicalLimit::GetDelimiter(input, offset_expression.get());
+		if (!val.is_null) {
+			offset = val.GetValue<idx_t>();
+		}
+		if (offset > 1ULL << 62ULL) {
+			throw BinderException("Max value %lld for LIMIT/OFFSET is %lld", offset, 1ULL << 62ULL);
+		}
+		state.is_offset_delimited = true;
+	}
+
+	if (!PhysicalLimit::HandleOffset(input, state.current_offset, 0, DConstants::INVALID_INDEX)) {
+		return SinkResultType::NEED_MORE_INPUT;
+	}
+
+	state.data.Append(input);
+	return SinkResultType::NEED_MORE_INPUT;
+}
+
+//===--------------------------------------------------------------------===//
+// Source
+//===--------------------------------------------------------------------===//
+class LimitPercentOperatorState : public GlobalSourceState {
+public:
+	LimitPercentOperatorState() : chunk_idx(0), limit(DConstants::INVALID_INDEX), current_offset(0) {
+	}
+
+	idx_t chunk_idx;
+	idx_t limit;
+	idx_t current_offset;
+};
+
+unique_ptr<GlobalSourceState> PhysicalLimitPercent::GetGlobalSourceState(ClientContext &context) const {
+	return make_unique<LimitPercentOperatorState>();
+}
+
+void PhysicalLimitPercent::GetData(ExecutionContext &context, DataChunk &chunk, GlobalSourceState &gstate_p,
+                                   LocalSourceState &lstate) const {
+	auto &gstate = (LimitPercentGlobalState &)*sink_state;
+	auto &state = (LimitPercentOperatorState &)gstate_p;
+	auto &limit_percent = gstate.limit_percent;
+	auto &offset = gstate.offset;
+	auto &limit = state.limit;
+	auto &current_offset = state.current_offset;
+
+	if (gstate.is_limit_percent_delimited && limit == DConstants::INVALID_INDEX) {
+		idx_t count = gstate.data.Count();
+		limit = MinValue((idx_t)(limit_percent / 100 * count), count);
+		if (limit == 0) {
+			return;
+		}
+	}
+
+	if (state.chunk_idx >= gstate.data.ChunkCount() || current_offset >= limit + offset) {
+		return;
+	}
+
+	DataChunk &input = gstate.data.GetChunk(state.chunk_idx);
+	if (PhysicalLimit::HandleOffset(input, current_offset, offset, limit)) {
```

Comment:
Does this work if the offset is bigger than the vector size?

e.g.

```sql
SELECT * FROM range(10000) LIMIT 0.1% OFFSET 8000;
```

Reply:
Great, I will fix it, Thanks!

Reply:
You're right, there is a problem. I will fixt it.